src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -662,6 +667,7 @@ void StorageObjectStorageQueue::commit(
 
     ProfileEvents::increment(ProfileEvents::ObjectStorageQueueCommitRequests, requests.size());
 
+    std::lock_guard lock(startup_mutex);
```

Comment:
We can remove mutex from here, because this code cannot be executed without startup being called